### PR TITLE
[1918] Fixed to show previous allocations

### PR DESCRIPTION
**Pull request by defong**
### Context
Allocations and accredited provider and its training providers
### Changes proposed in this pull request
Previously the list of training providers (for making `repeat` request type) has actual relationship with its training providers. Based on that they are running a PE course for next year as will as they have previously being allocated.

Now the same list will be based solely on previously being allocated.

### Guidance to review
Commits are probably best to view as 2 parts with the first 3 of making the changes and its related tests. The latter 3 as make the changes downstream.

You need to actually spin this up, and use https://github.com/DFE-Digital/teacher-training-api/pull/1972 

Allocations
-  is a closed loop system
- it has a sliding window of time
  - aka `journey mode`
  - it has it own lifecycle from `open`  -> `closed` -> `confirmed`
- it has its own request_type, `initial` (first timer), `repeat` (subsequence), `declined` (dead end)
- it spawns across two allocation cycles, 
  - the `previous` which holds the list of training providers that is encouraged to be used for this `current`
  - the list of training providers (for making `repeat` request type)
    - may not be related to the accredited body due to time preceptive so may not be an existing provider or still an training provider

#### Notes
It is a bit fragile as new allocations can be made with non existing training provider

#### Before
image
#### After
image

#### Before
image

#### After
image

#### Before
image

#### After
image

image

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by defong** [comment]
[Allocations page 'tweak'](https://trello.com/c/4tNE2tay/1918-allocations-page-tweak)

File: spec/controllers/providers/allocations_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+require "rails_helper"
+
+RSpec.describe Providers::AllocationsController, type: :controller do
```
Comment (gpeng) [inline spec/controllers/providers/allocations_controller_spec.rb:3]: Aren't we duplicating a lot testing here? We have a feature spec which we prefer over these.

**Review by gpeng — COMMENTED** [review]
Looks good. Not sure we need the controller spec.

File: spec/controllers/providers/allocations_controller_spec.rb (hunk 1 above)
Comment (defong) [reply spec/controllers/providers/allocations_controller_spec.rb:3]: No, they test all the low hanging fruits that is not tested in the feature spec.

Such as which partial template is expected to be rendered, and the training providers that are all expected to be.

**Review by gpeng — APPROVED** [review]

**Pull request merged**